polygeist/tools/cgeist/Lib/CGExpr.cc

```diff
@@ -870,6 +870,26 @@ ValueCategory MLIRScanner::VisitConstructCommon(clang::CXXConstructExpr *Cons,
 
 ValueCategory
 MLIRScanner::VisitArraySubscriptExpr(clang::ArraySubscriptExpr *Expr) {
+  LLVM_DEBUG({
+    llvm::dbgs() << "VisitArraySubscriptExpr: ";
+    Expr->dump();
+    llvm::dbgs() << "\n";
+  });
+
+  assert(!Expr->getBase()->getType()->isVLSTBuiltinType() &&
+         "Not supported yet");
+
+  if (Expr->getBase()->getType()->isVectorType()) {
+    // Handle the vector case.
+    ValueCategory Base{Visit(Expr->getBase()).getValue(Builder), false};
```

Comment:
This line should be replaced with:

```cpp
    auto Base = Visit(Expr->getBase());
```

when we implement obtaining lvalues from expressions as this one.